arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -8,7 +8,6 @@
 
 config ARM_MPU
 	bool "ARM MPU Support"
-	depends on CPU_HAS_MPU
```

Comment:
I understand that this Kconfig file is sourced, only if CPU_HAS_MPU.
Still, I don't see how this removal is helping.
And also, from a formal point of view, it is more "correct" to keep the HW capability dependency on an arch-build option.

Reply:
@ioannisg 
See https://github.com/zephyrproject-rtos/zephyr/pull/14134 for a longer explanation.

There is no such thing as a "conditional" source in Kconfig. I'd be impossible to implement.

All `if` (and `depends on` on menus) really is is syntactic sugar for adding a bunch of `depends on` to the stuff within.